leader-election-impl/src/main/java/com/palantir/leader/health/LocalCorruptionDetector.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.leader.health;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import com.palantir.common.concurrent.NamedThreadFactory;
+import com.palantir.common.concurrent.PTExecutors;
+
+public class LocalCorruptionDetector {
+    private final ScheduledExecutorService executor = PTExecutors.newSingleThreadScheduledExecutor(
+            new NamedThreadFactory("corruption-detector", true));
+
+    public LocalCorruptionDetector() {
+        scheduleWithFixedDelay();
+    }
+
+    public boolean isCorruptionOnLocal() {
+        return corruptionOnLocal;
+    }
+
+    private boolean corruptionOnLocal = false;
+
+    private void scheduleWithFixedDelay() {
+        executor.scheduleWithFixedDelay(() -> {
+                    if (detectedSignsOfCorruption()) {
+                        corruptionOnLocal = true;
+                    }
+                },
+                300,
+                300,
+                TimeUnit.SECONDS);
```

Comment:
Two notes:
- I'd suggest using a constant to express the duration here
- I'd prefer writing 5 minutes rather than 300 seconds (unless there's some hidden importance to 300 seconds I'm missing).